script/constructron.lua

```diff
@@ -354,107 +332,73 @@ end
 
 ---@param job Job
 me.graceful_wrapup = function(job)
-    for k, value in pairs(global.job_bundles[job.bundle_index]) do
-        if not value.returning_home and not (value.action == "clear_items") and not (value.action == "retire") and not (value.action == "check_build_chunk") and not (value.action == "check_decon_chunk") then
+    local allowed_actions = {
+        "clear_items",
+        "retire",
+        "check_build_chunk",
+        "check_decon_chunk",
+        "check_upgrade_chunk"
+    }
+    for k, value in pairs(global.job_bundles[job.bundle_index]) do -- clear unwanted actions
+        if not value.returning_home and not allowed_actions[value.action] then
             global.job_bundles[job.bundle_index][k] = nil
         end
     end
     local new_t = {}
     local i = 1
-    for _,v in pairs(global.job_bundles[job.bundle_index]) do
+    for _, v in pairs(global.job_bundles[job.bundle_index]) do -- reindex the job_bundle
         new_t[i] = v
         i = i + 1
     end
     global.job_bundles[job.bundle_index] = new_t
 end
 
 ---@param job Job
----@return boolean?

```

Comment:
You still return a boolean from this function. This annotation was correct to have.